linker.go

```diff
@@ -457,3 +458,33 @@ func resolveKconfigReferences(insns asm.Instructions) (_ *Map, err error) {
 
 	return kconfig, nil
 }
+
+func resolveKsymReferences(insns asm.Instructions) error {
```

Comment:
I had a few comments on this function, so I played around with it a bit and made it closer to the style of the code base:
```go

func resolveKsymReferences(insns asm.Instructions) error {
	var missing []string

	iter := insns.Iterate()
	for iter.Next() {
		ins := iter.Ins
		meta, _ := ins.Metadata.Get(ksymMetaKey{}).(*ksymMeta)
		if meta == nil {
			continue
		}

		v := meta.VarSpec
		if v.Addr != 0 {
			ins.Constant = int64(v.Addr)
			continue
		}

		if meta.Binding == elf.STB_WEAK {
			ins.Constant = 0
			continue
		}

		name := v.Var.TypeName()
		if !slices.Contains(missing, name) {
			missing = append(missing, name)
		}
	}

	if len(missing) > 0 {
		return fmt.Errorf("missing ksyms: %s", strings.Join(missing, ","))
	}

	return nil
}
```

- shorter lines and shorter var names
- moving things closer to where they're used
- don't check for tautological conditions, e.g. (a == 0 followed by a != 0)
- inverted slices.Contains, it would append all occurrences otherwise